### PR TITLE
Revised contributing.md, edited docs

### PR DESCRIPTION
**Pull request by nimraahmed**
### Changes made:

- Revised contributing.md as per discussion
- renamed workflows.mdx to contributing.mdx
- Added brief description for what reader should expect from each page of the doc
- Fixed broken/incorrect links in the doc
- Added missing links
- rearranged existing glossary terms alphabetically
- rearranged the explanation of the module architecture according to the provided directory in the docs

### CODE REVIEW

File: docs/package-lock.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
         "clsx": "^1.2.1",
         "graphiql": "^2.4.7",
         "graphql": "^16.6.0",
+        "iframe-resizer-react": "^1.1.0",
```
Comment (FelixMalfait) [inline docs/package-lock.json:17]: We should remove this file and use yarn.lock exclusively not npm

Comment (FelixMalfait) [reply docs/package-lock.json:17]: Maybe add package-lock.json to gitignore by the way since yarn is used across the repo?

**Review by FelixMalfait — APPROVED** [review]
Great work

File: docs/package-lock.json (hunk 1 above)
Comment (nimraahmed) [reply docs/package-lock.json:17]: i was removing and caching it while i solved the issue with yarn on my machine (solved it now) but i'll add it to gitignore too

File: docs/.gitignore [hunk 2]
```diff
@@ -18,3 +18,5 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+package-lock.json
```
Comment (FelixMalfait) [inline docs/.gitignore:22]: You've added it to gitignore but haven't removed the file from the tree. Adding to gitignore doesn't remove the file

**Pull request merged**